regulation/cms-0057-f/bulk-export-client-service/utils.bal

```diff
@@ -0,0 +1,344 @@
+        log:printError("Error occurred while getting the status.", statusResponse);
+    }
+    return null;
+}
+
+# Write file into file system.
+#
+# + downloadLink - file location  
+# + fileName - file name
+# + return - error if failed
+public isolated function saveFileInFS(string downloadLink, string fileName) returns error? {
+
+    http:Client statusClientV2 = check new (downloadLink);
+    stream<byte[], io:Error?> streamer = check getFileAsStream(downloadLink, statusClientV2) ?: new ();
+
+    check io:fileWriteBlocksFromStream(fileName, streamer);
+    check streamer.close();
+    log:printDebug(string `Successfully downloaded the file. File name: ${fileName}`);
+}
+
+# Send file from file system to a file server via ftp.
+#
+# + config - server config
+# + sourcePath - file path
+# + fileName - file name
+# + return - error if failed
+public isolated function sendFileFromFSToFTP(TargetServerConfig config, string sourcePath, string fileName) returns error? {
+    // Implement the FTP server logic here.
+    ftp:Client fileClient = check new ({
+        host: config.host,
+        auth: {
+            credentials: {
+                username: config.username,
+                password: config.password
+            }
+        }
+    });
+    stream<io:Block, io:Error?> fileStream
+        = check io:fileReadBlocksAsStream(sourcePath, 1024);
+    check fileClient->put(string `${config.directory}/${fileName}`, fileStream);
+    check fileStream.close();
+}
+
+# Util method to handle file download.
+#
+# + exportSummary - metadata of the export
+# + exportId - assigned export id, (local reference)
+# + return - error if failed
+public isolated function downloadFiles(json exportSummary, string exportId) returns error? {
+
+    ExportSummary exportSummary1 = check exportSummary.cloneWithType(ExportSummary);
+
+    foreach OutputFile item in exportSummary1.output {
+        log:printDebug("Downloading the file.", url = item.url);
+        error? downloadFileResult = saveFileInFS(item.url, string `${clientServiceConfig.targetDirectory}${file:pathSeparator}${exportId}${file:pathSeparator}${item.'type}-exported.ndjson`);
+        if downloadFileResult is error {
+            log:printError("Error occurred while downloading the file.", downloadFileResult);
+        }
+        if targetServerConfig.'type == "ftp" {
+            // download the file to the FTP server
+            // implement the FTP server logic
+            error? uploadFileResult = sendFileFromFSToFTP(targetServerConfig, string `${clientServiceConfig.targetDirectory}${file:pathSeparator}${item.'type}-exported.ndjson`, string `${item.'type}-exported.ndjson`);
+            if uploadFileResult is error {
+                log:printError("Error occurred while sending the file to ftp.", downloadFileResult);
+
+            }
+        }
+    }
+    lock {
+        boolean _ = updateExportTaskStatusInMemory(taskMap = exportTasks, exportTaskId = exportId, newStatus = "Downloaded");
```

Comment:
plug the function for the relevant implementation 
Refer: https://github.com/wso2/open-healthcare-prebuilt-services/pull/85/files#r1957674248